rotkehlchen/usage_analytics.py

```diff
@@ -22,134 +24,123 @@
+        data = response.json()
+        date = deserialize_timestamp_from_date(
+            date=data['commit']['commit']['author']['date'],
+            formatstr='%Y-%m-%dT%H:%M:%S',
+            location='Analytics',
         )
-    except requests.exceptions.RequestException:
-        return None
-
-    if response.status_code != HTTPStatus.OK:
-        return None
-
+    except requests.exceptions.RequestException as e:
+        log.debug(f'Failed to get metadata information for geoip file. {str(e)}')
+        metadata_query_failed = True
+    except DeserializationError as e:
+        log.debug(f'Failed to deserialize date in metadata information for geoip file. {str(e)}')
+        metadata_query_failed = True
+    except KeyError as e:
+        log.debug(f'Github response for geoip file had missing key {str(e)}')
+        metadata_query_failed = True
+
+    if metadata_query_failed:
+        old_files = list(geoip_dir.glob('*.mmdb'))
+        if len(old_files) == 0:
+            return None
+        filename = old_files[0]
+    else:
+        filename = geoip_dir / f'geoip-{date}.mmdb'
+
+    if not filename.is_file():
+        # Remove old files
+        files = geoip_dir.glob('*.*')
+        for f in files:
+            os.remove(f)
+        # Download latest version
+        try:
+            response = requests.get(
+                url='https://github.com/geoacumen/geoacumen-country/raw/master/Geoacumen-Country.mmdb',  # noqa: E501
+                timeout=DEFAULT_TIMEOUT_TUPLE,
+                stream=True,
+            )
+        except requests.exceptions.RequestException as e:
+            log.debug(f'Failed to download geoip database file. {str(e)}')
+            return None
+        with open(filename, 'wb') as outfile:
+            response.raw.decode_content = True
+            shutil.copyfileobj(response.raw, outfile)
+
+    # get user ip
     try:
-        json_ret = jsonloads_dict(response.text)
-    except JSONDecodeError:
-        return None
-
-    return GeolocationData(
-        country_code=json_ret.get('country', 'unknown'),
-        city=json_ret.get('city', 'unknown'),
-    )
-
-
-def query_ipstack() -> Optional[GeolocationData]:
-    """
-    10,000 requests per month tied to the API Key
-    https://ipstack.com/
-    """
-    try:
-        response = requests.get(
-            'http://api.ipstack.com/check?access_key=affd920d6e1008a614900dbc31d52fa6',
-            timeout=LOCATION_DATA_QUERY_TIMEOUT,
-        )
-    except requests.exceptions.RequestException:
-        return None
-
-    if response.status_code != HTTPStatus.OK:
+        u = miniupnpc.UPnP()
+        u.discoverdelay = 200
+        u.discover()
+        u.selectigd()
+        user_ip = u.externalipaddress()
+    except Exception as e:  # pylint: disable=broad-except
+        # In my case I was getting a generic exception rised
+        # when failing to detect the ip
+        log.debug(f'Failed to get ip for analytics. {str(e)}')
```

Comment:
Don't write things like this in a comment. Tell me that in discord or as a comment in your own PR. 
```suggestion
        # can raise base exception so we catch it
        log.debug(f'Failed to get ip via UPnP for analytics. {str(e)}')
```